app/agent/runtime/qemu/qemu.service.test.ts

```diff
@@ -0,0 +1,387 @@
+        brService.createContainer(osIm, "osCt"),
+        brService.createContainer(extraIm1, "extraCt1"),
+        brService.createContainer(extraIm1, "extraCt2"),
+        brService.createContainer(extraIm2, "extraCt3"),
+      ]);
+      // We will make 6 mounts, including 2 readonly ones
+      const [osB, extraB1, extraB2, extraB3, extraB4, extraB5] = await waitForPromises(
+        [osCt, extraCt1, extraCt2, extraCt3, extraIm1, extraIm2].map((what) =>
+          brService.expose(what, EXPOSE_METHOD.BLOCK_DEV),
+        ),
+      );
+      const instance = injector.resolve(Token.QemuService)(runId);
+      await instance.withRuntime(
+        {
+          ssh: {
+            username: "root",
+            password: "root",
+          },
+          vcpuCount: 1,
+          memSizeMib: 128,
+          fs: {
+            "/": osB,
+            "/mount1rw": extraB1,
+            "/mount2rw": extraB2,
+            "/mount3rw": extraB3,
+            "/mount4ro": extraB4,
+            "/very/deeply/nested/non/existing/ro/mount/point": extraB5,
+          },
+          cleanupAfterStop: true,
+          shouldPoweroff: true,
+        },
+        async ({ ssh }) => {
+          const mountInfo = await (
+            await execSshCmd({ ssh }, ["cat", "/proc/self/mountinfo"])
+          ).stdout;
+          // Assert mount options
+          expect(mountInfo).toMatch(/^.*?\/.*?ext4 .*? rw,discard$/gm);
+          expect(mountInfo).toMatch(/^.*?\/mount1rw.*?ext4 .*? rw,discard$/gm);
+          expect(mountInfo).toMatch(/^.*?\/mount2rw.*?ext4 .*? rw,discard$/gm);
+          expect(mountInfo).toMatch(/^.*?\/mount3rw.*?ext4 .*? rw,discard$/gm);
+          expect(mountInfo).toMatch(/^.*?\/mount4ro.*?ext4 .*? ro$/gm);
+          expect(mountInfo).toMatch(
+            /^.*?\/very\/deeply\/nested\/non\/existing\/ro\/mount\/point.*?ext4 .*? ro$/gm,
+          );
+          // Assert mount contents :)
+          const im1Content = "This is layer 2a";
+          const im2Content = "This is layer 2b";
+          await waitForPromises(
+            [
+              ["/mount1rw/fileBA", im1Content],
+              ["/mount2rw/fileBA", im1Content],
+              ["/mount3rw/fileBB", im2Content],
+              ["/mount4ro/fileBA", im1Content],
+              ["/very/deeply/nested/non/existing/ro/mount/point/fileBB", im2Content],
+            ].map(([path, content]) =>
+              expect(execSshCmd({ ssh }, ["cat", path])).resolves.toHaveProperty("stdout", content),
+            ),
+          );
+        },
+      );
+    })(),
+);
+
+test.concurrent.each(testCases)(
+  `If requested then keeps running in the background on %s`,
+  async (_name, remoteTag) =>
+    provideBlockRegistry(async ({ injector, runId, brService }) => {
+      const osIm = await brService.loadImageFromRemoteRepo(remoteTag, "osIm");
+      const osCt = await brService.createContainer(osIm, "osCt");
+      const osB = await brService.expose(osCt, EXPOSE_METHOD.BLOCK_DEV);
```

Comment:
shouldn't you hide it later?

Reply:
It's handled globally: https://github.com/hocus-dev/hocus/pull/93/files#diff-6728b317769e3b07f2f0926e1b27c3dadcd3958b1eb186a33b3cc8b3234f6047R162